Upgrade

iky opened the pull request:
* A few formatting improvements
  * Apply [black](https://github.com/ambv/black)
  * trim trailing white spaces
  * sort imports
  * new PEP8s
* Add support for Python 3.7 and remove support for Python 3.4
* Make tests working with both major Nameko versions - https://github.com/iky/nameko-slack/commit/d34f628639b6e7fdbf53fe0c9f610e7baf2b7c43

iky committed:
Add support for Python 3.7 and remove support for Python 3.4

iky committed:
A few formatting improvements

* trim trailing white spaces
* sort imports
* PEP8s...

iky committed:
Make tests working with both major Nameko versions

iky committed:
Apply black

iky committed:
Merge branch 'upgrade' into upgrade-nameko-3

iky committed:
Support black

iky committed:
Use Ubuntu Xenial on CI

mattbennett reviewed (commented):
Might want to consider adding a `.pre-commit.yaml` config file and/or enforcing black in CI.

Check out nameko/nameko-grpc for an example project set up that way.

iky committed:
Remove Python nightly CI support

iky committed:
package setup - make universial bdist wheel

iky committed:
package setup - isort

iky committed:
package setup - flake8

iky committed:
package setup - add pro-commit

iky committed:
package setup - add black to pre-commit

iky committed:
package setup - run pre-commit checks as part of the tests (including ci)

iky committed:
Better test variable naming

iky committed:
Run static checks only once and on Python 3.6

iky committed:
Run latest nameko tests first

iky committed:
Run latest python first

iky committed:
Use main and latest isort package

iky committed:
Only pre-commit needed for static stage

iky committed:
Save some travis effort ...

...by replacing matrix expansion with explicit job definitions

mattbennett commented on .travis.yml:
```diff
@@ -1,27 +1,44 @@
+  - python: 3.6
+    env: NAMEKO_VERSION===2.8.5
+  - python: 3.6
+    env: NAMEKO_VERSION="==2.8.1 pytest<3.3.0"
```
`DEPS=`? ;)

iky replied:
:+1:  good suggestion .)

mattbennett commented:
One other thought related to this --

It might be nice to have a general "prerelease" item in the matrix, so you are always running against the latest prerelease version, rather than hardcoding to a specific version.

With a general "prerelease" item you could also add a cron-based execution to Travis, which would alert early if a prerelease version of Nameko introduces incompatibilities.

iky committed:
Nicer CI matrix setup

iky commented:
@mattbennett I replaced the hardcoded prerealase with generic prerelease installation. I also added a cronjob running daily but I marked the prerelease test as allow-failure so it does not fail builds with tests passing against normal nameko releases.

iky commented:
I changed the cronjob to weekly .)
